plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/sale-badge/edit.tsx

```diff
@@ -4,47 +4,37 @@
 import { useBlockProps } from '@wordpress/block-editor';
 import type { BlockEditProps } from '@wordpress/blocks';
 import { ProductQueryContext as Context } from '@woocommerce/blocks/product-query/types';
-import { useEffect } from '@wordpress/element';
+import { store as editorStore } from '@wordpress/editor';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import Block from './block';
 import type { BlockAttributes } from './types';
-import { useIsDescendentOfSingleProductTemplate } from '../shared/use-is-descendent-of-single-product-template';
 
 const Edit = ( {
 	attributes,
-	setAttributes,
 	context,
 }: BlockEditProps< BlockAttributes > & { context: Context } ): JSX.Element => {
 	const blockProps = useBlockProps();
 
+	const isTemplate = useSelect( ( select ) => {
+		const { getCurrentPostType } = select( editorStore );
+
+		const template = getCurrentPostType() === 'wp_template';
+
+		return template;
+	}, [] );
+
```

Comment:
It's too generic and applies to Product Catalog as well which results in On Sale displayed on every product in Product Collection. It should only apply:

<img width="1035" alt="image" src="https://github.com/user-attachments/assets/d6180109-c064-4a4e-8e33-7cc6a7306beb" />

I think we can keep:

```
const { isDescendentOfSingleProductTemplate } =
		useIsDescendentOfSingleProductTemplate();
```

but simply avoiding setting it as an `attribute`.

Reply:
Nice catch! I completely to forgot about the Product Collection!

Reply:
Fixed with 838f988